### PR TITLE
Add use case: Filter resources

### PR DESCRIPTION
**Pull request by alien-mcl**
As promised, filtering use case.

### CODE REVIEW

File: drafts/use-cases/8.filtering-events.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+So I can find events I'm interested in.
+
+As an API consumer
+I want to be able to filter resources
+So I can find those interesting to me.
+
+### Usage
+```javascript
+var client = new HydraClient();
+var data = client.get("/api/events?schema:eventName=.*test.*");
```
Comment (lanthaler) [inline drafts/use-cases/8.filtering-events.md]: Same as on the other PR I would suggest to extend this a bit

File: drafts/use-cases/8.filtering-events.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+From technical point of view, this behavior could be achieved
+with special purpose virtual predicates 
+(i.e.:  imaginatory hydra:limit and hydra:offset).
+The argument against is that these are not direct properties 
+of resources being filtered.
+Supporting argument would be that similar approach is used 
+in i.e. MS SQL pagination queries, where the row
+(which can be a one-to-one relation to an entity in our cases)
+is extended with a special purpose ROW_NUMBER value
```
Comment (lanthaler) [inline drafts/use-cases/8.filtering-events.md]: The problem is that not all backends would be able to implement. So we would need to provide some mechanisms to advertise support for it.

Comment (asbjornu) [inline drafts/use-cases/8.filtering-events.md]: Each database engine basically has its own implementation of paging. There's `take`, `limit`, `top`, `skip`, etc. Hydra should be agnostic to how the serialization of this looks like and instead focus on the abstract concept, no?

Comment (RubenVerborgh) [inline drafts/use-cases/8.filtering-events.md]: We might want to postpone this for now; let's not set such difficult paging decisions in stone before we actually have completed paging.

Comment (tpluscode) [inline drafts/use-cases/8.filtering-events.md]: These are connected concepts. Again, it's why I oppose putting too much into the core vocabulary. We need a minimal set of extension point to support any paging/filtering scheme. And if you ask me, the current `PartialCollactionView` comes very close.

Comment (alien-mcl) [inline drafts/use-cases/8.filtering-events.md]: That's why I put it in the considerations. I also agree about not having too much in the core.
Still, currently spec neither have that covered nor allows any extension other than generic IriTemplate (which may end with stumbling upon the issue #118).

**Pull request merged**